sdcm/utils/decode_kallsyms.py

```diff
@@ -0,0 +1,48 @@
+import bisect
+import re
+import subprocess
+
+
+def binary_search_recursive(array, element):
+    return bisect.bisect_right(array, element, key=lambda y: int(y.split()[0], 16)) - 1
+
+
+
+def cut_addresses_from_log_line_kernel_callstack(line):
+    my_regex = re.compile('0x([0-9a-f]{16})')
+    return my_regex.findall(line)
+
+
+def get_lines_from_log(file_path):
+    file_content = dict()
+    with open(file_path, 'r') as file:
+        for line in file:
+            if ('kernel callstack' in line and '0x' in line) or 'Reactor stalled for' in line:
+                file_content[line] = cut_addresses_from_log_line_kernel_callstack(line)
+    return file_content
+
+
+def get_kallsyms(path_to_kallsyms):
+    with open(path_to_kallsyms, 'r') as file:
+        file_content = file.readlines()
+    return file_content
+
+
+def append_content_to_file(file_path, content):
+    with open(file_path, 'a') as file:
+        file.write(content + '\n')
+
+
+# TODO: dynamically get the 3 files below (from command line arguments for example where results could have a default)
+results_file = 'results.log'
+file_content = get_kallsyms('kallsyms')
+open(results_file, 'w').close()
+
+for full_line, stall in get_lines_from_log('system.log').items():
+    append_content_to_file(results_file, '############################################################################')
```

Comment:
The line separator could look like `'#' * 76` and defined before the loop

Reply:
done